maven-plugin/src/main/java/org/openmrs/maven/plugins/Run.java

```diff
@@ -73,19 +74,26 @@ public Run(AbstractTask other, String serverId){
 	 */
 	private Boolean watchApi;
 
+	private ServerHelper serverHelper;
+
 	public void executeTask() throws MojoExecutionException, MojoFailureException {
 		if (serverId == null) {
 			File currentProperties = Server.checkCurrentDirForServer();
 			if (currentProperties != null) serverId = currentProperties.getName();
 		}
 		serverId = wizard.promptForExistingServerIdIfMissing(serverId);
 		Server server = loadValidatedServer(serverId);
-		if (port != null && port != 8080) {
-			server.setParam("tomcat.port", String.valueOf(port));
-		}
-		else{
-			server.setParam("tomcat.port", "8080");
-		}
+
+		port = Integer.valueOf(server.getPort());
+		String message = "Set server port or use default from parameters";
+		this.requestPortChnage(message, server);
```

Comment:
Please see this task. Maybe I didn't got it right but there is written that I should ask for port every time. Thanks for answer. `https://issues.openmrs.org/browse/SDK-199`